authorizations/authorization-jdbc/src/main/java/org/apache/gravitino/authorization/jdbc/JdbcSQLBasedAuthorizationPlugin.java

```diff
@@ -0,0 +1,456 @@
+      throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    onUserAdded(user);
+
+    for (Role role : roles) {
+      onRoleCreated(role);
+      String sql = getRevokeRoleSQL(role.name(), "USER", user.name());
+      executeUpdateSQL(sql);
+    }
+    return true;
+  }
+
+  @Override
+  public Boolean onGrantedRolesToGroup(List<Role> roles, Group group)
+      throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    onGroupAdded(group);
+
+    for (Role role : roles) {
+      onRoleCreated(role);
+      String sql =
+          getGrantRoleSQL(role.name(), "USER", String.format("%s%s", GROUP_PREFIX, group.name()));
+      executeUpdateSQL(sql);
+    }
+    return true;
+  }
+
+  @Override
+  public Boolean onRevokedRolesFromGroup(List<Role> roles, Group group)
+      throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    onGroupAdded(group);
+
+    for (Role role : roles) {
+      onRoleCreated(role);
+      String sql =
+          getRevokeRoleSQL(role.name(), "USER", String.format("%s%s", GROUP_PREFIX, group.name()));
+      executeUpdateSQL(sql);
+    }
+    return true;
+  }
+
+  @Override
+  public Boolean onUserAdded(User user) throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    String sql = getCreateUserSQL(user.name());
+    executeUpdateSQL(sql);
+    return true;
+  }
+
+  @Override
+  public Boolean onUserRemoved(User user) throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    String sql = getDropUserSQL(user.name());
+    executeUpdateSQL(sql);
+    return true;
+  }
+
+  @Override
+  public Boolean onUserAcquired(User user) throws AuthorizationPluginException {
+    throw new UnsupportedOperationException("Doesn't support to acquired a user");
+  }
+
+  @Override
+  public Boolean onGroupAdded(Group group) throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    String name = String.format("%s%s", GROUP_PREFIX, group.name());
+    String sql = getCreateUserSQL(name);
+    executeUpdateSQL(sql);
+    return true;
+  }
+
+  @Override
+  public Boolean onGroupRemoved(Group group) throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    String name = String.format("%s%s", GROUP_PREFIX, group.name());
+    String sql = getDropUserSQL(name);
+    executeUpdateSQL(sql);
+    return true;
+  }
+
+  @Override
+  public Boolean onGroupAcquired(Group group) throws AuthorizationPluginException {
+    throw new UnsupportedOperationException("Doesn't support to acquired a group");
+  }
+
+  @Override
+  public Boolean onOwnerSet(MetadataObject metadataObject, Owner preOwner, Owner newOwner)
+      throws AuthorizationPluginException {
+    beforeExecuteSQL();
+
+    if (newOwner.type() == Owner.Type.USER) {
+      onUserAdded(new TemporaryUser(newOwner.name()));
+    } else if (newOwner.type() == Owner.Type.GROUP) {
+      onGroupAdded(new TemporaryGroup(newOwner.name()));
+    } else {
```

Comment:
I think we can use `GroupEntity` instead `TemporaryGroup`, and use `UserEntity` instead `TemporaryUser`.
It's best not to introduce a new concept.

Reply:
Removed TemporaryUser/Group.